src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -215,6 +215,32 @@ public async Task InitializeAsync(IEnumerable<FunctionMetadata> functions, Cance
             }
         }
 
+        public async Task<IDictionary<string, WorkerStatus>> GetWorkerStatusesAsync()
+        {
+            var workerChannels = (await GetInitializedWorkerChannelsAsync()).ToArray();
+            _logger.LogDebug($"[HostMonitor] Checking worker statuses (Count={workerChannels.Length})");
+
+            // invoke the requests to each channel in parallel
```

Comment:
Chose to sample all the channels. We could also decide to sample just one, and assume that load balancing will mean that sample is representative...

Reply:
I prefer the implementation you have here. Sampling all workers to ensure scaling does kick even if one of the workers in under load. As of now, invocations distribution is simple round robin. Eventually invocation distribution can take advantage for WorkerStatus. But for now, what you have is better.